### PR TITLE
Intellij cleanups

### PR DESCRIPTION
**Pull request by rhowe-gds**
Some of these are maybe more risky than the usual straightforward ones..

Lots of things have changed access levels and lots of things have become `final`.
The tests pass for me locally...

### CODE REVIEW

File: src/main/java/uk/gov/pay/products/config/ProductsModule.java [hunk 1]
```diff
@@ -65,16 +65,18 @@ private JpaPersistModule jpaModule(ProductsConfiguration configuration) {
         final Properties properties = new Properties();
         properties.put("javax.persistence.jdbc.driver", dbConfig.getDriverClass());
         properties.put("javax.persistence.jdbc.url", dbConfig.getUrl());
-        properties.put("javax.persistence.jdbc.user", dbConfig.getUser());
-        properties.put("javax.persistence.jdbc.password", dbConfig.getPassword());
+        if (dbConfig.getUser() != null)
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/products/config/ProductsModule.java]: We like curly braces around our `if` statements!

If you wanted to do it all on one line, you could do this:

```java
Optional.ofNullable(dbConfig.getUser).ifPresent(user -> properties.put("javax.persistence.jdbc.user", user));
```

File: src/main/java/uk/gov/pay/products/service/PaymentCreator.java [hunk 2]
```diff
@@ -91,6 +90,7 @@ private PaymentEntity mergePaymentEntityWithoutReferenceCheck(PaymentEntity paym
     }
 
     private PaymentEntity mergePaymentEntityWithReferenceNumberCheck(PaymentEntity paymentEntity) {
+        int MAX_NUMBER_OF_RETRY_FOR_UNIQUE_REF_NUMBER = 3;
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/products/service/PaymentCreator.java]: This was fine where it was but should be a `private static final int`

File: src/main/java/uk/gov/pay/products/validations/RequestValidations.java [hunk 3]
```diff
@@ -116,8 +115,4 @@
             return false;
         };
     }
-
-    public static Function<JsonNode, Boolean> isNotBoolean() {
-        return jsonNode -> !ImmutableList.of("true", "false").contains(jsonNode.asText().toLowerCase());
-    }
```
Comment (alexbishop1) [inline src/main/java/uk/gov/pay/products/validations/RequestValidations.java]: Presumably this method is unused

Comment (rhowe-gds) [reply src/main/java/uk/gov/pay/products/validations/RequestValidations.java]: That's certainly what IntelliJ and `git grep` think

**Pull request merged**